src/vni-manager.js

```diff
@@ -0,0 +1,156 @@
+ * If the vni for the specified vnid does not exist, a new one will be created.
+ *
+ * @this node unless otherwise specified on an API
+ * @param componentFacade a component facade
+ *
+ * @return the component facade, extended to include the vni interface
+ * 
+ * Usage examples: 
+ *     var vni = node.vni(); // retrieve the default VNI 
+ *     var vni = node.vni(vnid); // retrieve the vni associated with the vnid
+ *
+ *     node.deleteAllVnis();
+ *     node.deleteVni( vnid );
+ *     node.vni(vnid).delete();
+ *
+ *     node.vni().errorState( state );
+ *     var state = node.vni().errorState();
+ *
+ *     node.vni().inputStates( {input: state} );
+ *     var inputStates = node.vni().inputStates();
+ *
+ *     node.vni().outputState( state );
+ *     var state = node.vni().outputState();
+ */
+module.exports = function( componentFacade ) { 
+
+    _.extend( componentFacade, 
+        {
+
+          /**
+           * Delete all vnis assiciated with the node instance.
+           *
+           * @this node instance
+           * @return node instance node for easy chaining
+           */ 
+          deleteAllVnis: function() { 
+              // reassign vnis array to an empty array and let garbage collection clean up
+              this.vnis = {};
+              return this;
+          },
+
+          /**
+           * Delete the vni associated with the specified VNID
+           * 
+           * @param vnid vnid identifying which vni to delete
+           *
+           * @this node instance
+           * @return node instance node for easy chaining
+           */ 
+          deleteVni: function( vnid ) {
+
+              if ( vnid in this.vnis ) {
+                  delete this.vnis[vnid];
+              } 
+
+              return this;
+          },
+
+          /**
+           * Getter for a vni facade specified by the vnid parameter.  If no VNID is specified,
+           * the IIP VNID will be used.  If the VNI does not yet exist, a new VNI will be
+           * created.
+           * 
+           * @this node 
+           * @param vnid vnid identifying which vni to delete
+           *
+           * @return node for easy chaining
+           */ 
+          vni: function( vnid ) { 
+
+              vnid = vnid || DEFAULT_VNID;
+
+              if ( _.isUndefined( this.vnis[vnid] )  ) {
+
+                  // have no vni so create one
+                  this.vnis[vnid] = {}; 
+
+                  // TODO: Add parentVni setting here
+                  // TODO: Add previousLms here
+              }
+
+              return {
+                  delete: _.bind( this.deleteVni, this, vnid ),
+                  inputStates: _.partial( inputStates, this, vnid ), 
+                  errorState: _.partial( errorState, this.vnis[vnid] ), 
+                  outputState: _.partial( outputState, this.vnis[vnid] ) 
+              };
+          },
+
+          vnis: {}
```

Comment:
Added a test that demonstrates that the function results in an extended component facade with the new vni-manager apis.